nautobot/utilities/filters.py

```diff
@@ -358,6 +366,27 @@ def get_filters(cls):
         """
         filters = super().get_filters()
 
+        content_type = None
+        model_filterset = None
+        if hasattr(cls._meta.model, "_meta"):
+            content_type = get_content_type_string(cls._meta.model)
+            model_filterset = get_filterset_for_model(cls._meta.model)
+            # TODO: Verify if this bruteforce approach is the best way
+            if cls not in FILTER_CLASS_STRINGS:
+                FILTER_CLASS_STRINGS.append(cls)
+
+        if registry["plugin_filter_extensions"].get(content_type) and cls is model_filterset:
+            for filter_extension in registry["plugin_filter_extensions"][content_type]:
+                if not filter_extension().filterset():
+                    continue
+                for filter_name, filter in filter_extension().filterset().items():
```

Comment:
With the above suggestion this would become something like:

```suggestion
        if registry["plugin_filterset_extensions"].get(content_type) and cls is model_filterset:
            for filterset_extension in registry["plugin_filterset_extensions"][content_type]:
                for filter_name, filter in filterset_extension.items():
```